### ISSUE TITLE
Cyclical Learning Rates for Training Neural Networks

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#37]
# Information
### URL
https://arxiv.org/abs/1506.01186
### Authors
Leslie N. Smith
### Date of issue
4 Apr 2017
### Related Websites
https://sgugger.github.io/how-do-you-find-a-good-learning-rate.html
https://sgugger.github.io/the-1cycle-policy.html#the-1cycle-policy
### Package
https://github.com/davidtvs/pytorch-lr-finder
### Code
# Reading Motivation
kaggleでfast.aiを使ったコードを見ていて, lrの決め方が気になったため.
# Summary
cyclical learning rates という自動で学習率を決める手法の提案. 単調に学習率を下げるのではなく, boundary 間で周期的に変化させる. 学習率を固定させた場合と比べ, 画像分類タスクにおいて, チューニング無し&少ないイテレーション回数で性能の向上.　またboundary を決めるための LR range test を提案. これは少ないepoch 数で, lrを単調に増加させていき, accuracy が減少したところを max_lr とするという手法. Adaptive  系のoptimizer との併用が可能で, 異なるデータセット, モデルで性能の向上を確認.

# Proposed Method
base_lr, max_lr, stepsize を決めて, 学習率を振動させる. 
![Screenshot from 2022-02-13 16-29-28](https://user-images.githubusercontent.com/39965001/153743485-a53ee2e0-f0f7-459c-a638-f30165f74dd9.png)

その他の提案
* 周期ごとにmax_lr を半減(?)
* base_lr, max_lr にexponential factorを掛けて iteration 毎に小さくしていく.
![Screenshot from 2022-02-13 16-30-51](https://user-images.githubusercontent.com/39965001/153743523-fef33ff4-6997-4bfe-a06e-337b7fae5baf.png)

LR range test
base_lr, max_lr を決めるために, lr を大きくしながら数epoch学習し, accuracy が上がり始めるところを base_lr, accuracy が下がったところを max_lr とする. Fig3 では base_lr = 0.001, max_lr = 0.006.
![Screenshot from 2022-02-13 16-35-03](https://user-images.githubusercontent.com/39965001/153743667-d7ad7292-0da8-481b-8859-3a264a945e99.png)
